src/lib/tvgShape.cpp

```diff
@@ -48,6 +48,34 @@ unique_ptr<Shape> Shape::gen() noexcept
     return unique_ptr<Shape>(new Shape);
 }
 
+unique_ptr<Shape> Shape::duplicate() const noexcept
+{
+    auto shape = Shape::gen();
```

Comment:
I think you just copy method like this.
shape->pImpl->duplicate(IMPL);  //here duplicate() copies all properties.

Shape::Impl::duplicate(Shape::Impl* from)
{
   x = from->x;
   y = from->y;
   memcpy() and etc;
   flag = from->flag;
}